### ISSUE TITLE
Error/Crash with kmsdrm on Ubuntu 22.04 + Wayland + Raspberry 4b : wl_drm@20 error!

### ISSUE THREAD

**Issue by Fredrum** [libsdl-org/SDL#5743]
Hello!

I'm adding Ubuntu compatibility to my program and I got it more or less working ok on the Desktop.
But the program also supports running in the tty screen, rendering using **kmsdrm**.

This is now causing error+crash and I'm not sure what to do about it.

Setup is:
SDL 2.0.20 built from your published source release.
Raspberry Ubuntu 22.04 - with Wayland.
Raspberry Pi 4b
gnu c/c++

The error message is like this:
```
wl_drm@20 error: authenticate failed
SDL could not create window: Could not initialize EGL
```

Here's my code that sets up SDL window+render:
https://github.com/Fredrum/chiaki/blob/dev_rpi02/rpi/src/main.cpp

Does anyone know what this means and what I might be able to do about it?

**UPDATE**:   I found a tip for how to work around it but I'd still love to be able to fix it in the code!
If I do this the error goes away and my program runs fine under tty,
`WAYLAND_DISPLAY=   path/my_program`
Basically setting that env var to nothing, then running my program.

It seems like Wayland is still making itself known under tty, unlike X11!?

Got it from:  https://www.reddit.com/r/swaywm/comments/jn8mxr/how_do_i_prevent_a_waylandnative_application_from/

Cheers!

**Comment by Fredrum** [comment]
Closing it something fixed it.
